pyClient/zeth/encryption.py

```diff
@@ -19,6 +19,55 @@
 from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat,\
     PublicFormat, NoEncryption
 
+# pylint: disable=line-too-long
+
+# References:
+#
+# \[Bernstein06]
+#  "Curve25519:new Diffie-Hellman speed records"
+#  Daniel J. Bernstein,
+#  International Workshop on Public Key Cryptography, 2006,
+#  <https://cr.yp.to/ecdh/curve25519-20060209.pdf>
+#
+# \[LangleyN18]
+#  "Chacha20 and poly1305 for ietf protocols."
+#  Adam Langley and Yoav Nir,
+#  RFC 8439, 2018,
+#  <https://tools.ietf.org/html/rfc8439>
+
+# This implementation makes use of the `cryptography` library with OpenSSL
+# backend. For the avoidance of doubt, the implementation adheres to the
+# appropriate standards as follows. (links refer to specific versions of external
+# libraries, to ensure that line numbers are correct, but the descriptions are
+# expected to hold for all versions.)
+
+# As described in [Bernstein06], private keys may be generated as 32 random
+# bytes with bits 0, 1 and 2 of the first byte cleared, bit 7 of the last byte
+# cleared, and bit 6 of the last byte set. This happens at key generation
+# time. See:
+#
+#   https://github.com/openssl/openssl/blob/be9d82bb35812ac65cd92316d1ae7c7c75efe9cf/crypto/ec/ecx_meth.c#L81
```

Comment:
Note, the execution path through the code is very difficult to follow. To be sure, I've also experimentally verified that keys are generated with the expectd properties.